tests/fetch/test_cds.py

```
@@ -0,0 +1,513 @@
+    @patch("indra.fetch.cds._get_thread_s3_client")
+    @patch("indra.fetch.cds.generate_kerchunk_index")
+    @patch("indra.fetch.cds._download_file", return_value=True)
+    def test_full_pipeline_uploads_and_returns_uploaded(self, mock_dl, mock_ki, mock_s3_factory, tmp_path):
+        """Happy path: download → index → .nc upload → .json upload → stage='uploaded'."""
+        filename = "era5_land_2t_2024_01.nc"
+        output_dir, kerchunk_dir = self._setup_dirs(tmp_path, filename)
+
+        mock_s3 = MagicMock()
+        mock_s3_factory.return_value = mock_s3
+
+        fname, success, stage = _process_single_file(
```

Ruff (0.15.9)

[warning] Unpacked variable `fname` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)